tests/base/quadrature_push_back.cc

```diff
@@ -0,0 +1,46 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2017 - 2018 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+
+
+#include <deal.II/base/quadrature.h>
+
+#include "../tests.h"
+
+/*
+ * Test that it is possible to add a point to the quadrature formula using the
+ * push_back function.
+ */
+template <int dim>
+void
+test_push_back()
+{
+  Quadrature<dim>  quadrature;
+  const Point<dim> point;
+  const double     weight = 1;
+  quadrature.push_back(point, weight);
+  AssertThrow(1 == quadrature.size(), ExcInternalError());
```

Comment:
I think you also want to check the location of that point, and the value of the weight.

Reply:
Wrote the point and weight to the log.